pkg/storage/filesystem/device.go

```diff
@@ -83,6 +95,83 @@ func (d *DeviceInfo) IsPXEPartition() bool {
 	return d.Label == "ZOSPXE"
 }
 
+func (d *DeviceInfo) IsPartitioned() bool {
+	return len(d.Children) != 0
+}
+
+func (d *DeviceInfo) GetUnallocatedSpaces(ctx context.Context) ([]DiskSpace, error) {
+	args := []string{
+		"--json", d.Path, "unit", "B", "print", "free",
+	}
+	output, err := exec.CommandContext(ctx, "parted", args...).CombinedOutput()
+	if err != nil {
+		return nil, fmt.Errorf("parted command error: %w", err)
+	}
+
+	var diskData struct {
+		Disk struct {
+			Partitions []DiskSpace `json:"partitions"`
+		} `json:"disk"`
+	}
+	if err := json.Unmarshal(output, &diskData); err != nil {
+		return nil, fmt.Errorf("failed to parse parted output: %v", err)
+	}
+
+	validSpaces := []DiskSpace{}
+	for _, part := range diskData.Disk.Partitions {
+		if isValidAsDevice(part) {
+			validSpaces = append(validSpaces, part)
+		}
+	}
+
+	return validSpaces, nil
+}
+
+func (d *DeviceInfo) AllocateEmptySpace(ctx context.Context, space DiskSpace) error {
+	args := []string{
+		d.Path, "mkpart", "primary", space.Start, space.End,
```

Comment:
this command is only valid for DOS Disklabels (DOS Partition table)
for GPT disklabels it's `mkpart $somename $fstype space.start space.end`

Reply:
https://github.com/diskfs/go-diskfs
instead of shelling into parted commands, this would be **a lot** more manageable

Reply:
the code already handle the `zos-cache` named devices, this part for preparing all possible spaces zos can use for cache if no `zos-cache`  partition is found.

Reply:
i tired to give the pkg a try but no way to get the empty space on a disk, so will end up using shell command for that, then parse and convert the output to use with go-diskfs pkg, i see the current state are more simpler